docs/02-providers/aws/02-iam.md

```diff
@@ -28,4 +82,246 @@ provider:
+
+functions:
+  func0: # will assume 'myCustRole0'
+    # declare one of the following...
+    role: 'myCustRole0'                                                # defining 'role' will override any 'roleArn' declaration // must validly reference a role defined in the service
+    roleArn: { 'Fn::GetAtt': ['myCustRole0', 'Arn'] }                  # must validly reference a role defined in the service
+    roleArn: 'arn:aws:iam::0123456789:role//my/cust/path/myCustRole0'  # must validly reference a role defined in your account
+    ...
+  func1: # will assume 'myDefaultRole'
+    ...    # does not define roleArn
+
+resources:
+  Resources:
+    myDefaultRole:
+      Type: AWS::IAM::Role
+      RoleName: MyDefaultRole
+      Properties:
+        Path: /my/default/path
+        AssumeRolePolicyDocument:
+          Version: 2012-10-17
+          Statement:
+            - Effect: Allow
+              Principal:
+                Service:
+                  - lambda.amazonaws.com
+                Action: sts:AssumeRole
+        Policies:
+          - PolicyName: myPolicyName
+            PolicyDocument:
+              Version: 2012-10-17
+              Statement:
+                - Effect: Allow # note that these rights are given in the default policy and are required if you want logs out of your lambda(s)
+                  Action:
+                    - logs:CreateLogGroup
+                    - logs:CreateLogStream
+                    - logs:PutLogEvents
+                  Resource: arn:aws:logs:${region}:${accountId}:log-group:/aws/lambda/*:*:*
+                -  Effect: "Allow"
+                   Action:
+                     - "s3:PutObject"
+                   Resource:
+                     Fn::Join:
+                       - ""
+                       - - "arn:aws:s3:::"
+                         - "Ref" : "ServerlessDeploymentBucket"
+    myCustRole0:
+      Type: AWS::IAM::Role
+      RoleName: MyCustRole0
+      Properties:
+        Path: /my/cust/path
+        AssumeRolePolicyDocument:
+          Version: 2012-10-17
+          Statement:
+            - Effect: Allow
+              Principal:
+                Service:
+                  - lambda.amazonaws.com
+                Action: sts:AssumeRole
+        Policies:
+          - PolicyName: myPolicyName
+            PolicyDocument:
+              Version: 2012-10-17
+              Statement:
+                - Effect: Allow
+                  Action:
+                    - logs:CreateLogGroup
+                    - logs:CreateLogStream
+                    - logs:PutLogEvents
+                  Resource: arn:aws:logs:${region}:${accountId}:log-group:/aws/lambda/*:*:*
+                - Effect: Allow
+                  Action:
+                    - ec2:CreateNetworkInterface
+                    - ec2:DescribeNetworkInterfaces
+                    - ec2:DetachNetworkInterface
+                    - ec2:DeleteNetworkInterface
+                  Resource: *
```

Comment:
Need to be wrapped in "" so that YAML won't throw an exception --> Resource: "*"